src/utils/disk_utils.hpp

```diff
@@ -25,4 +25,14 @@ inline std::optional<std::string> GetOldDiskKeyOrNull(storage::Delta *head) {
   return std::nullopt;
 }
 
+inline uint64_t GetEarliestTimestamp(storage::Delta *head) {
+  if (head == nullptr) return 0;
+  uint64_t ts = head->timestamp->load(std::memory_order_acquire);
+  while (head->next != nullptr) {
+    head = head->next;
+    ts = std::min(ts, head->timestamp->load(std::memory_order_acquire));
```

Comment:
why min? why not just iterate until you found the last delta? Last delta should always be earliest

Reply:
Agree